src/visualizer/graphPanel/GraphPanel.cpp

```diff
@@ -0,0 +1,17 @@
+//
+// Created by stefan on 23.07.2023.
+//
+
+#include "GraphPanel.hpp"
+
+namespace age::visualizer {
+GraphPanel::GraphPanel(QWidget* pParent) noexcept : QWidget(pParent) {}
+
+void GraphPanel::mousePressEvent(QMouseEvent* pEvent) {
+  auto mousePos = pEvent->position();
+  cds::UniquePointer<Vertex> vertex = cds::makeUnique<Vertex>((int) mousePos.x(), (int) mousePos.y(), this);
```

Comment:
Use auto;
Create QPointF to remove the warnings for "reduntant casting".
```suggestion
  auto vertex = cds::makeUnique<Vertex>(mousePos.x(), mousePos.y(), this);
```
Alternatively, if casting is required, do not use C Style casting. Use `static_cast`
```suggestion
  auto vertex = cds::makeUnique<Vertex>((int) mousePos.x(), (int) mousePos.y(), this);
```